### PR TITLE
*: log format for server

### PR DESCRIPTION
**Pull request by nolouch**
Signed-off-by: nolouch <nolouch@gmail.com>

### What problem does this PR solve? 
Format the log according to our RFC.

### What is changed and how it works?
This PR is going to do the log format for `server` packages.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: server/cluster_info.go [hunk 1]
```diff
@@ -482,7 +491,10 @@ func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
 	// Mark isNew if the region in cache does not have leader.
 	var saveKV, saveCache, isNew bool
 	if origin == nil {
-		log.Debugf("[region %d] Insert new region {%v}", region.GetID(), core.HexRegionMeta(region.GetMeta()))
+		log.Debug("Insert new region",
+			zap.Uint64("region-id", region.GetID()),
+			zap.Any("meta-region", core.HexRegionMeta(region.GetMeta())),
```
Comment (rleungx) [inline server/cluster_info.go]: How about using `Reflect`?

File: server/leader.go [hunk 2]
```diff
@@ -348,7 +354,7 @@ func (s *Server) watchLeader(leader *pdpb.Member, revision int64) {
 // ResignLeader resigns current PD's leadership. If nextLeader is empty, all
 // other pd-servers can campaign.
 func (s *Server) ResignLeader(nextLeader string) error {
-	log.Infof("%s tries to resign leader with next leader directive: %v", s.Name(), nextLeader)
+	log.Info("tries to resign leader to next leader", zap.String("from", s.Name()), zap.String("to", nextLeader))
```
Comment (disksing) [inline server/leader.go]: tries -> try

File: server/join.go [hunk 3]
```diff
@@ -176,14 +177,14 @@ func PrepareJoinCluster(cfg *Config) error {
...
-		log.Error("failed to open:", err)
+		log.Error("failed to open", zap.Error(err))
 		return false
 	}
 	defer dir.Close()
 
 	names, err := dir.Readdirnames(-1)
 	if err != nil {
-		log.Error("failed to list:", err)
+		log.Error("failed to list", zap.Error(err))
```
Comment (disksing) [inline server/join.go]: failed to list directory?

File: server/join.go [hunk 4]
```diff
@@ -176,14 +177,14 @@ func PrepareJoinCluster(cfg *Config) error {
 func isDataExist(d string) bool {
 	dir, err := os.Open(d)
 	if err != nil {
-		log.Error("failed to open:", err)
+		log.Error("failed to open", zap.Error(err))
```
Comment (disksing) [inline server/join.go]: failed to open directory?

File: server/heartbeat_stream_test.go [hunk 5]
```diff
@@ -113,13 +114,13 @@ func newRegionheartbeatClient(c *C, grpcClient pdpb.PDClient) *regionHeartbeatCl
 
 func (c *regionHeartbeatClient) close() {
 	if err := c.stream.CloseSend(); err != nil {
-		log.Errorf("Failed to terminate client stream: %v", err)
+		log.Error("Failed to terminate client stream", zap.Error(err))
```
Comment (disksing) [inline server/heartbeat_stream_test.go]: Failed -> failed

File: server/coordinator.go [hunk 6]
```diff
@@ -221,7 +222,7 @@ func (c *coordinator) run() {
 	scheduleCfg.Schedulers = scheduleCfg.Schedulers[:k]
 	c.cluster.opt.store(scheduleCfg)
 	if err := c.cluster.opt.persist(c.cluster.kv); err != nil {
-		log.Errorf("can't persist schedule config: %v", err)
+		log.Error("can't persist schedule config", zap.Error(err))
```
Comment (disksing) [inline server/coordinator.go]: can't -> cannot

File: server/cluster_info.go [hunk 7]
```diff
@@ -492,18 +504,32 @@ func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
...
+				zap.Uint64("new-confver", r.GetConfVer()),
+			)
 			saveKV, saveCache = true, true
 		}
 		if region.GetLeader().GetId() != origin.GetLeader().GetId() {
 			if origin.GetLeader().GetId() == 0 {
 				isNew = true
 			} else {
-				log.Infof("[region %d] Leader changed from store {%d} to {%d}", region.GetID(), origin.GetLeader().GetStoreId(), region.GetLeader().GetStoreId())
+				log.Info("Leader Changed",
```
Comment (disksing) [inline server/cluster_info.go]: leader changed

File: server/cluster_info.go [hunk 8]
```diff
@@ -482,7 +491,10 @@ func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
 	// Mark isNew if the region in cache does not have leader.
 	var saveKV, saveCache, isNew bool
 	if origin == nil {
-		log.Debugf("[region %d] Insert new region {%v}", region.GetID(), core.HexRegionMeta(region.GetMeta()))
+		log.Debug("Insert new region",
```
Comment (disksing) [inline server/cluster_info.go]: Insert -> insert

File: server/cluster_info.go [hunk 9]
```diff
@@ -492,18 +504,32 @@ func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
...
+				zap.Uint64("region-id", region.GetID()),
+				zap.String("detail", core.DiffRegionKeyInfo(origin, region)),
+				zap.Uint64("old-version", o.GetVersion()),
+				zap.Uint64("new-version", r.GetVersion()),
+			)
 			saveKV, saveCache = true, true
 		}
 		if r.GetConfVer() > o.GetConfVer() {
-			log.Infof("[region %d] %s, ConfVer changed from {%d} to {%d}", region.GetID(), core.DiffRegionPeersInfo(origin, region), o.GetConfVer(), r.GetConfVer())
+			log.Info("Region ConfVer Changed",
```
Comment (disksing) [inline server/cluster_info.go]: region ConfVer changed

File: server/cluster_info.go [hunk 10]
```diff
@@ -492,18 +504,32 @@ func (c *clusterInfo) handleRegionHeartbeat(region *core.RegionInfo) error {
 			return ErrRegionIsStale(region.GetMeta(), origin.GetMeta())
 		}
 		if r.GetVersion() > o.GetVersion() {
-			log.Infof("[region %d] %s, Version changed from {%d} to {%d}", region.GetID(), core.DiffRegionKeyInfo(origin, region), o.GetVersion(), r.GetVersion())
+			log.Info("Region Version Changed",
```
Comment (disksing) [inline server/cluster_info.go]: region version changed

File: server/grpc_service.go [hunk 11]
```diff
@@ -652,9 +653,12 @@ func (s *Server) UpdateGCSafePoint(ctx context.Context, request *pdpb.UpdateGCSa
 		if err := s.kv.SaveGCSafePoint(newSafePoint); err != nil {
 			return nil, err
 		}
-		log.Infof("updated gc safe point to %d", newSafePoint)
+		log.Info("updated gc safe point suces",
```
Comment (huachaohuang) [inline server/grpc_service.go]:
```suggestion
		log.Info("updated gc safe point",
```

**Pull request merged**